### PR TITLE
Organization Functionalities

### PR DESCRIPTION
**Pull request by nam-t24**
- Create org
- View your created org
- Update org
- Show Team members
- Installed shadcn for some components so run `npm install` again

Screenshot 2024-02-27 at 2 24 29 PM

Sidebar organization tabs will not show if user is not part of org so user can only click on overview to join org

Screenshot 2024-02-27 at 2 25 30 PM

If user is not part of org they will be taken to /dashboard/organization/overview/orgSignUp page
There are also error messages if user tries to submit empty required fields

Screenshot 2024-02-27 at 2 27 58 PM

Once user joins org they can see org data with teams. Other org sidebar tabs will be shown as well. Only owner can edit org info. Button and input areas will disappear

### CODE REVIEW

File: app/dashboard/organization/overview/page.tsx [hunk 1]
```diff
@@ -1,11 +1,213 @@
...
+                    }
+
+                </div>
+                {/* Team Section */}
+                <OrgTeam orgTeam={orgTeam || []} isOwner = {isOwner || false}/>
+            </div>
+            {/* End Org info section */}
+            <div className="mt-24">
+                <div className="text-2xl">Danger Zone</div>
+                <div className="w-96 border-b-[1px] border-body mt-1 mb-3"></div>
```
Comment (Andy581) [inline app/dashboard/organization/overview/page.tsx]: Little nitpick that we can include in the next pr or just ignore. I think it looks better if we change mb-3 to mb-5
image
image

File: components/dashboard/TeamMemberCard.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+            <div>
+                <div className="flex items-center space-x-2 > * + * select-none">
+                    <div>{name || ""}</div>
+                    {index == 0 && <div className='text-xs bg-red-800/20 border-[1px] border-red-900 text-red-900 rounded-xl px-3'>Owner</div>}
+                </div>
+                <div className="text-body text-sm">{email}</div>
+
+            </div>
+            {isOwner && <button className='text-2xl'>
+                <MoreVertIcon fontSize='inherit'/>
```
Comment (Andy581) [inline components/dashboard/TeamMemberCard.tsx:14]: Are we gonna give the MoreVertIcon any functionality for this pr?

File: app/dashboard/organization/overview/page.tsx [hunk 3]
```diff
@@ -1,11 +1,213 @@
...
-            <PageHeading header="Overview" description="Your are currently not part of any organization"/>
+            <PageHeading header="Overview" description="Your organization"/>
+            {loading ? 
+            <div className="text-center 2xl:mt-72 mt-52 text-darkmaroon">
+                <div className="text-xl font-medium">Getting organization data</div>
+                <div className="loadingAnimation"><div></div><div></div><div></div></div>
+            </div> 
+            :
+            <div>
+            <div className="flex items-stretch mt-8 2xl:space-x-6 > * + * space-x-4 > * + * animate-in">
```
Comment (Andy581) [inline app/dashboard/organization/overview/page.tsx:138]: What does > * + * do? I can't find any documentation about it in Tailwind.

File: components/dashboard/TeamMemberCard.tsx (hunk 2 above)
Comment (nam-t24) [reply components/dashboard/TeamMemberCard.tsx:14]: Nah cuz this pr already too massive. I'll put it in another feature pr

File: app/dashboard/organization/overview/page.tsx (hunk 3 above)
Comment (nam-t24) [reply app/dashboard/organization/overview/page.tsx:138]: Space between

File: app/dashboard/organization/overview/page.tsx (hunk 1 above)
Comment (nam-t24) [reply app/dashboard/organization/overview/page.tsx]: will do wizard nd

**Review by Andy581 — APPROVED** [review]

**Pull request merged**